internal/status/route_status.go

```diff
@@ -45,40 +48,42 @@ func ParseOptionsFromMetadata(options []UpdateOptions, bulk bool) ([]string, map
 					continue
 				}
 
-				ingress := ingressArr[0] + "/" + ingressArr[1]
-				option.IngSvc = ingress
-				if opt, ok := updateIngressOptions[ingress]; ok {
+				ingressIPKey := ingressns + "/" + option.Vip
+				option.IngSvc = ingressns
+				if opt, ok := updateIngressOptions[ingressIPKey]; ok {
 					for _, hostname := range option.ServiceMetadata.HostNames {
 						if !utils.HasElem(opt.ServiceMetadata.HostNames, hostname) {
 							opt.ServiceMetadata.HostNames = append(opt.ServiceMetadata.HostNames, option.ServiceMetadata.HostNames...)
-							updateIngressOptions[ingress] = opt
+							updateIngressOptions[ingressIPKey] = opt
 						}
 					}
 				} else {
-					updateIngressOptions[ingress] = option
+					updateIngressOptions[ingressIPKey] = option
 				}
 			}
 		} else {
 			// insecure VSes, servicemetadata comes from Pools.
 			ingress := option.ServiceMetadata.Namespace + "/" + option.ServiceMetadata.IngressName
+			ingressIPKey := ingress + "/" + option.Vip
 			option.IngSvc = ingress
-			if opt, ok := updateIngressOptions[ingress]; ok {
+			if opt, ok := updateIngressOptions[ingressIPKey]; ok {
 				for _, hostname := range option.ServiceMetadata.HostNames {
 					if !utils.HasElem(opt.ServiceMetadata.HostNames, hostname) {
 						opt.ServiceMetadata.HostNames = append(opt.ServiceMetadata.HostNames, option.ServiceMetadata.HostNames...)
-						updateIngressOptions[ingress] = opt
+						updateIngressOptions[ingressIPKey] = opt
 					}
 				}
 			} else {
-				updateIngressOptions[ingress] = option
+				updateIngressOptions[ingressIPKey] = option
 			}
 		}
 	}
 
 	ingressesToUpdate := make([]string, len(updateIngressOptions))
 	i := 0
 	for k := range updateIngressOptions {
-		ingressesToUpdate[i] = k
+		kSlice := strings.Split(k, "/")
+		ingressesToUpdate[i] = kSlice[0] + "/" + kSlice[1]
```

Comment:
This is guaranteed to always yield 2 values? If not, we may through a nil pointer.